cpp/src/arrow/compute/kernels/scalar_cast_test.cc

```diff
@@ -1485,51 +1485,57 @@ TEST(Cast, StringToBoolean) {
 TEST(Cast, StringToInt) {
   for (auto string_type : {utf8(), large_utf8()}) {
     for (auto signed_type : {int8(), int16(), int32(), int64()}) {
-      CheckCast(ArrayFromJSON(string_type, R"(["0", null, "127", "-1", "0"])"),
-                ArrayFromJSON(signed_type, "[0, null, 127, -1, 0]"));
+      CheckCast(
+          ArrayFromJSON(string_type, R"(["0", null, "127", "-1", "0", "0x0", "0x7F"])"),
+          ArrayFromJSON(signed_type, "[0, null, 127, -1, 0, 0, 127]"));
```

Comment:
Add a test case for hex with negative sign (eg., `-0x34`) and another for multiple prefixed zeros (eg., `00000x7f`).

Reply:
Those cases are explicitly not allowed. Do you mean that I should add them to `CheckCastFails`?
A negative before a hex does not make sense. If the value can be negative (signed integer) then you should use the binary representation of that negative number, for example for int8 0xFF is -1 and 0xF0 is -128.
For the case of multiple prefixed zeros, I opted for not allowing that to be allowed, since allowing it would make the logic more complex (probably slower) and it seems like it would be a very odd way for a hex string to be formatted.

Reply:
Ohh, sorry...I meant as failing test cases. It is just a way to ensure more code coverage in tests.